packages/client/test/events.spec.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  OpenFeature,
+  ApiEvents,
+  ProviderEvents,
+  Provider,
+} from '../src';
+import EventEmitter from 'events';
+
+const ERROR_REASON = 'error';
+const ERROR_CODE = 'MOCKED_ERROR';
+
+const MOCK_EMITTER: EventEmitter = {
+  emit: jest.fn( () => {
+    return true;
+  }),
+  initialize: jest.fn( () => {
+    return MOCK_EMITTER;
+  })
+}as unknown as EventEmitter;
+
+const MOCK_ERROR_EMITTER: EventEmitter = {
+  emit: jest.fn( (event) => {
+    if (event == ProviderEvents.Ready) {
+      throw Error('Event error!');
+    }
+    return true;
+  }),
+  initialize: jest.fn( () => {
+    return MOCK_EMITTER;
+  })
+}as unknown as EventEmitter;
+
+const MOCK_PROVIDER: Provider = {
+  metadata: {
+    name: 'mock-events-success',
+  },
+  initialize: jest.fn(() =>  {
+    return Promise.resolve(undefined);
+  }),
+  events: MOCK_EMITTER,
+} as unknown as Provider;
+
+const ERROR_PROVIDER: Provider = {
+  metadata: {
+    name: 'mock-events-failure',
+  },
+  initialize: jest.fn(() =>  {
+    return Promise.resolve(undefined);
+  }),
+  events: MOCK_ERROR_EMITTER,
+} as unknown as Provider;
+
+describe('Events', () => {
+  // set timeouts short for this suite.
+  jest.setTimeout(1000);
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Requirement 5.1.1', () => {
+    it('The provider **MAY** define a mechanism for signalling the occurrence of an event`PROVIDER_READY`',
```

Comment:
Nitpick: `it('The provider **MAY**...` reads weird. I recommend something like ``it('defines a mechanism...` so the test output is more readable, we don't need to put the spec text in each `it()`, IMHO. Doesn't really matter though.